doc: Added nRF54L05 and nRF54L10 snippets (#18090)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

Developing with nRF54L Series
=============================

* Added:

>   * :ref:`nRF54l_snippets` to emulate these targets on an nRF54L15 DK.

Reviewer: @wbober  Do you think we need a statement to be added here that L15 emulated devices are only development purposes.

>   * :ref:`nRF54l_snippets` to emulate theses devices on an nRF54L15 DK.

Reviewer 2: ```suggestion
  * :ref:`nRF54l_snippets` to emulate devices on an nRF54L15 DK.
```

Author: These snippets will emulate only  nRF54L05 and nRF54L10. Hence retained the word "these".

>   * :ref:`nRF54l_snippets` to emulate theses devices on an nRF54L15 DK.

Reviewer 3: ```suggestion
  * :ref:`nRF54l_snippets` to emulate nRF54L05 and nRF54L10 on an nRF54L15 DK.
```

Author: I was trying to avoid mention of nRF54L05 and nRF54L10 twice in this sentence. Could you check the updated statement?

>   * The :ref:`ug_nrf54l_cryptography` page, providing more information about the cryptographic peripherals of the nRF54L Series devices, the programming model for referencing keys, and the configuration.

Reviewer 2: ```suggestion
  * The :ref:`ug_nrf54l_cryptography` page that provides more information about the cryptographic peripherals of the nRF54L Series devices, programming model for referencing keys, and configuration.
```
